wgpu-hal/src/dx12/adapter.rs

```diff
@@ -243,6 +244,7 @@ impl super::Adapter {
                 private_caps,
                 presentation_timer,
                 workarounds,
+                dx12_shader_compiler: dx12_shader_compiler.clone(),
```

Comment:
I just cloned this down the initialization pipeline from Instance to Device as it was simple, and I wasn't sure if Instance lasted longer than Device.

Reply:
It does, but cloning here is fine, this is slow code.